monitor/monitor.go

```diff
@@ -44,54 +45,53 @@ type Monitor struct {
 
 // Run starts monitoring.
 func (m *Monitor) Run(npages int) {
-	log.Info("Starting monitoring traffic")
-	c := bpf.DefaultPerfEventConfig()
-	c.NumPages = npages
-
-	me, err := bpf.NewPerCpuEvents(c)
-	if err != nil {
-		log.WithError(err).Error("Error while starting monitor")
-		return
-	}
-	monitorEvents = me
-
+	eventConfig.NumPages = npages
 	// Dump stat
 	t := time.NewTicker(5 * time.Second)
-	go func(t *time.Ticker) {
-		for {
-			select {
-			case <-t.C:
-				if monitorEvents != nil {
-					m.dumpStat()
-				}
-			}
-		}
-	}(t)
-
-	// Main event loop
 	for {
-		todo, err := monitorEvents.Poll(pollTimeout)
-		if err != nil {
-			log.WithError(err).Error("Error in Poll")
-			if err == syscall.EBADF {
-				break
+		select {
+		case <-t.C:
+			if monitorEvents != nil {
+				m.dumpStat()
 			}
 		}
-		if todo > 0 {
-			if err := monitorEvents.ReadAll(m.receiveEvent, m.lostEvent); err != nil {
-				log.WithError(err).Warn("Error received while reading from perf buffer")
-			}
+	}
+}
+
+func purgeInactiveListeners() {
+	meta := &payload.Meta{Size: uint32(0)}
+	metaBuf, err := meta.MarshalBinary()
+	if err != nil {
+		log.WithError(err).Fatal("meta encode")
```

Comment:
Can we safely continue here?

Reply:
We should continue will change it to a info.